kong/plugins/opentelemetry/handler.lua

```diff
@@ -123,7 +124,26 @@ function OpenTelemetryHandler:access(conf)
     root_span.parent_id = parent_id
   end
 
-  propagation_set(conf.header_type, header_type, root_span, "w3c")
+  -- Balancer span propagation
+  -- Here we propagate the span that identifies the "last" balancer try
+  -- This has to happen "in advance". The span will be activated (linked)
+  -- after the OpenResty balancer results are available
+  local new_span = ngx_ctx.last_try_balancer_span
+  -- if the span was not already created by another tracing plugin instance
+  if new_span == nil then
+    -- create the last-try balancer span
+    new_span = tracer.create_span(nil, {
+      span_kind = 3, -- client
+      parent = root_span,
+    })
+    ngx_ctx.last_try_balancer_span = new_span
+  end
+
+  -- new_span can be noop (e.g. when should_sample == false) which should not
+  -- be sampled nor propagated. In that case, propagate the root_span instead.
+  local propagated = new_span.is_recording ~= false and new_span or root_span
+  -- propagate the last-try balancer span
```

Comment:
I guess this is independent of the tracing system we are using. Could we subtract this part of the code? It could be a shared function.

Reply:
totally. I was thinking of doing this as part of the future refactoring that will move most of this logic in the PDK, what do you think?

Reply:
Actually, my original consideration was to make the PDK drive the propagation process. The plugin tells the PDK how to propagate in a specific format, the PDK decides when and whether to do the propagation.

Reply:
moving this code in "propagation.lua" for now